Automattic/woocommerce-payments
# refactor: checkout error handling

Fixes https://github.com/Automattic/woocommerce-payments/issues/7561
Fixes https://github.com/Automattic/woocommerce-payments/issues/7587

#### Changes proposed in this Pull Request









I ensured that the error(s) returned by the `createPaymentMethod` functionality is/are handled correctly.

I added some new UTs to the `PaymentProcessor`, so that it should hopefully be ok.


Before - shortcode checkout:
![Screenshot 2023-11-07 at 6 32 54 PM](https://github.com/Automattic/woocommerce-payments/assets/273592/f37bb0dd-1a8d-4d39-96a4-881b9bdacf78)

After - shortcode checkout:
![Screenshot 2023-11-07 at 6 33 31 PM](https://github.com/Automattic/woocommerce-payments/assets/273592/fac13c0a-f77a-4e3d-a523-ef039c86d415)

After - blocks checkout:
<img width="575" alt="Screenshot 2023-11-07 at 6 32 03 PM" src="https://github.com/Automattic/woocommerce-payments/assets/273592/ad346ebc-0fb6-4ff9-afaa-6d74ca2ac541">


#### Testing instructions





- Onboard with a JP account
- Navigate to the shortcode checkout page and attempt to pay with a test JCB card `3566002020360505`
- Attempt to checkout
- The error handling/message should be improved
- Perform the same steps on the blocks checkout
- The error handling should be improved as well (prior to these changes there was no error displayed)


-------------------


**Post merge**
